### PR TITLE
Cu 1da4brz make enemies appear with a random material of the enemy materials set

### PR DESCRIPTION
**Pull request by saulzini**
Uploaded textures for materials - Melee Grunt
Paths:
Textures =  Penteract/Assets/Textures/MeleeGrunt/
Materials = Penteract/Assets/Materials/MeleeGrunt/
Created MeleeGruntRandomMat_Level1.prefab
Created RangeGruntv2RandomMat_Level1.prefab

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-1da4brz Make Enemies appear with a random Material (of the Enemy Materials Set)](https://app.clickup.com/t/1da4brz)

File: Penteract/Assets/Scripts/RangedAI.cpp [hunk 1]
```diff
@@ -74,10 +76,6 @@ void RangedAI::Start() {
 	if (meshObj) {
 		ComponentBoundingBox* bb = meshObj->GetComponent<ComponentBoundingBox>();
 		bbCenter = (bb->GetLocalMinPointAABB() + bb->GetLocalMaxPointAABB()) / 2;
-		meshRenderer = meshObj->GetComponent<ComponentMeshRenderer>();
-		if (meshRenderer) {
-			noDmgMaterialID = meshRenderer->materialId;
-		}
```
Comment (JaimeBea) [inline Penteract/Assets/Scripts/RangedAI.cpp:80]: Are you sure this should be removed? The dissolve and hit animations aren't playing.

**Review by JaimeBea — CHANGES_REQUESTED** [review]

File: Penteract/Assets/Scripts/RangedAI.cpp (hunk 1 above)
Comment (saulzini) [reply Penteract/Assets/Scripts/RangedAI.cpp:80]: That one is being set on the SetRandomMaterial();  once we know the random material 
noDmgMaterialID = materials[position];

Comment (JaimeBea) [reply Penteract/Assets/Scripts/RangedAI.cpp:80]: Yes, but what about meshRenderer?

Comment (saulzini) [reply Penteract/Assets/Scripts/RangedAI.cpp:80]: You are right . Fixed

**Review by JaimeBea — APPROVED** [review]
Bugs found:
- When dying, the enemies will change to the default material

File: Penteract/Assets/Scripts/RangedAI.cpp [hunk 2]
```diff
@@ -768,3 +768,39 @@ void RangedAI::UpdateDissolveTimer() {
...
+			if (meshRenderer && meshRenderer->materialId) {
+				materials.push_back(meshRenderer->materialId);
+			}
+		}
+
+		
+		if (!materials.empty()) {
+			//Random distribution it cant be saved into global 
+			std::random_device rd;  //Will be used to obtain a seed for the random number engine
+			std::mt19937 gen(rd()); //Standard mersenne_twister_engine seeded with rd()
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/RangedAI.cpp:790]: why declare it every time this method is called. You could declare the random device and init the mersenne twister outside this method.

**Review by Mefiso — COMMENTED** [review]
is MeleeGruntRandomMat_Level1.prefab the final enemy to be used? In that case I would change the name to something shorter and replace the old prefab.

More importantly, I've seen some of the random materials that can appear are blank... review this please.

**Review by Mefiso — APPROVED** [review]

**Review by isaacfdez — APPROVED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by magalenyo — APPROVED** [review]
We're gonna need to create new prefabs with the new materials

**Pull request merged**